src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -1294,6 +1319,30 @@ else if (v instanceof ListValue)
             if(e instanceof PlayerEntity) ((PlayerEntity) e).getHungerManager().addExhaustion((int) NumericValue.asNumber(v).getLong());
         });
 
+        put("absorption",(e, v)->{
+            if(e instanceof PlayerEntity) ((PlayerEntity) e).setAbsorptionAmount((float) NumericValue.asNumber(v).getLong());
+        });
+
```

Comment:
There's no need to restrict setting absorption level to players only since all living entities have the capability to have absorption points.